frame/nfts/src/lib.rs

```diff
@@ -729,8 +744,9 @@ pub mod pallet {
 			let origin = ensure_signed(origin)?;
 			let check_owner = check_owner.map(T::Lookup::lookup).transpose()?;
 
-			Self::do_burn(collection, item, |collection_details, details| {
-				let is_permitted = collection_details.admin == origin || details.owner == origin;
+			Self::do_burn(collection, item, |_, details| {
```

Comment:
you probably dont need collection details, and in do_burn, you can fails/ensure one storage read before.

Reply:
same below, don't see the reason to keep unused argument